Automattic/wp-calypso
# Store: Add a proper error message and UI when the server returns no label rates

In the `Label Purchase` flow, when the merchant requested label rates correctly but no rates were returned (an "empty list" response is not the same as an "error" response), the UI would look like this:
![image37](https://user-images.githubusercontent.com/1715800/44926522-ffc72e80-ad51-11e8-8bbd-53d497e4a297.png)

This is specially bad with the International Labels feature because it's easy to put items into a box that's not compatible with USPS international (such as `Regional A/B` boxes). The best solution is to not allow the merchant to select those boxes, but that will require at least server changes. The next best solution is to improve the UX.

This is an example of an international shipment. The first package is a `Regional Rate` box (not valid), and the second package is just the item in its original packaging (valid):
<img width="474" alt="screen shot 2018-08-31 at 19 02 23" src="https://user-images.githubusercontent.com/1715800/44926650-719f7800-ad52-11e8-99b5-4153b8d73c51.png">

To test:
* You can test this in Calypso or in WP-Admin (by `npm link`ing this branch into `woocommerce-services`).
* Try the normal case (domestic, or international with a supported package type like a custom box) and see the UI keeps working as usual.
* Force one package to be "invalid" so it doesn't return rates. The easiest way to do it is to select a `Regional Rate Box` in an international shipment.
* Check that the error messages are satisfactory, that the empty `Dropdown` is disabled, and that you can't purchase labels until you change to a correct package.

Marking as `In Progress` while I fix the tests.
